wip-mariocao-adjust-mining-probability.md

```diff
@@ -0,0 +1,87 @@
+<pre>
+    WIP: WIP-0007
+    Layer: Consensus (hard fork)
+    Title: Adjust mining probability
+    Authors: Mario Cao <mario@witnet.foundation>
+             Gorka Irazoqui <gorka.irazoki@gmail.com>
+    Discussions-To: `#dev-general` channel on Witnet Community's Discord server
+    Status: Draft
+    Type: Standards Track
+    Created: 2021-01-26
+    License: BSD-2-Clause
+</pre>
+
+
+## Abstract
+
+This proposal adjusts the mining algorithm to limit the probability of non-reputed and non-active nodes mining a block.
+
+The goal of this proposal is to mitigate potential network abuses that could be carried by malicious users controlling a large number of nodes. This could pose a serious threat to the network availability and security.
+
+
+## Motivation and Rationale
+
+after more than 3 months of the Witnet mainnet launch, the network has faced episodes of a bigger transaction latency than expected. In periods with a mempool full of transactions, blocks not using the complete space appeared in the Witnet network. This reduces the efficiency of the system, thus increasing indirectly the transaction latency perceived by users. Under a more critical scenario, this problem could pose an availability threat in terms of a [denial-of-service attack][DoS], in which a large number of malicious nodes could be mining empty blocks or, even, endanger the network stability by affecting internal network metrics (e.g. Active Reputation Set).
```

Comment:
```suggestion
After more than 3 months since the Witnet mainnet launch, the network has faced episodes where transaction latency was higher than expected. During periods with a mempool full of transactions, blocks not using the complete block space appeared in the Witnet block chain. This reduces the throughput of the system, thus indirectly increasing the transaction latency perceived by users. Under a more critical scenario, may a large number of malicious nodes intentionally mine empty blocks, this problem could pose an availability threat in terms of a [denial-of-service attack][DoS] or even endanger the network stability by affecting internal network metrics (e.g. Active Reputation Set).
```